src/Files/Views/LayoutModes/ColumnViewBase.xaml.cs

```diff
@@ -382,11 +382,21 @@ private async void FileList_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
             else if (e.Key == VirtualKey.Up || e.Key == VirtualKey.Down)
             {
                 // If list has only one item, select it on arrow down/up (#5681)
-                if (!IsItemSelected && FileList.Items.Count == 1)
+                if (!IsItemSelected)
                 {
                     FileList.SelectedIndex = 0;
                     e.Handled = true;
                 }
+
+                // Open slected directory
+                if (!IsRenamingItem && !ParentShellPageInstance.NavToolbarViewModel.IsEditModeEnabled)
+                {
+                    if (IsItemSelected && SelectedItem.PrimaryItemAttribute == StorageItemTypes.Folder)
+                    {
+                        ItemInvoked?.Invoke(new ColumnParam { NavPathParam = (SelectedItem is ShortcutItem sht ? sht.TargetPath : SelectedItem.ItemPath), ListView = FileList }, EventArgs.Empty);
```

Comment:
This does not work as it prevents moving selection, should be done in FileList_SelectionChanged